### ISSUE TITLE
harvest stop does not stop pollers that have been renamed

### ISSUE THREAD

**Issue by cgrinds** [NetApp/harvest#20]
Steps to reproduce - will add
1. Edit `harvest.yml` and add/enable one poller, call it `foo`
2. Verify not running 
```
bin/harvest status
Datacenter            Poller                PID        PromPort        Status
+++++++++++++++++++++ +++++++++++++++++++++ ++++++++++ +++++++++++++++ ++++++++++++++++++++
nane                  foo                                              not running
+++++++++++++++++++++ +++++++++++++++++++++ ++++++++++ +++++++++++++++ ++++++++++++++++++++
```
3. Start poller
```
bin/harvest start
Datacenter            Poller                PID        PromPort        Status
+++++++++++++++++++++ +++++++++++++++++++++ ++++++++++ +++++++++++++++ ++++++++++++++++++++
nane                  foo                   5828                       running
+++++++++++++++++++++ +++++++++++++++++++++ ++++++++++ +++++++++++++++ ++++++++++++++++++++
```
4. Edit `harvest.yml` change name of `foo` to `foo2`
5. Status fails because the "wrong" poller is queried
```
bin/harvest status
Datacenter            Poller                PID        PromPort        Status
+++++++++++++++++++++ +++++++++++++++++++++ ++++++++++ +++++++++++++++ ++++++++++++++++++++
nane                  foo2                                             not running
+++++++++++++++++++++ +++++++++++++++++++++ ++++++++++ +++++++++++++++ ++++++++++++++++++++
```
If you run `harvest start` you will create a new poller named foo2 while the first started poller is still running

```
ps aux | grep poller
root      5828  3.0  0.0 2795344 76752 ?       Sl   11:17   0:04 bin/poller --poller foo --loglevel 2 --promPort  --daemon
root      5912 49.8  0.0 2869588 97988 ?       Sl   11:19   0:02 bin/poller --poller foo2 --loglevel 2 --promPort  --daemon
```

start/stop/status should be more resilient to name changes. In a few places, we already interrogate /proc, extract command line arguments, and parse them. We should do the same for stop/status too. In other words, stopping and status should not depend on the names in harvest.yml, they instead should query the OS.

**Comment by rahulguptajss** [comment]
merged to Release branch. This can be tested now.

**Comment by sridevimm** [comment]
Validated the fix in harvest-21.06.1121-nightly build. 

After renaming the poller, it stopped running. After starting, it's running without any issues. Marking this issue as status/done.


[root@scspa2385980001 harvest]# bin/harvest status
Datacenter            Poller                PID        PromPort        Status              
+++++++++++++++++++++ +++++++++++++++++++++ ++++++++++ +++++++++++++++ ++++++++++++++++++++
umeng_aff300a         umeng_aff300a                                    not running         
F2240-127-26          F2240-127-26          27762      12991           running             
+++++++++++++++++++++ +++++++++++++++++++++ ++++++++++ +++++++++++++++ ++++++++++++++++++++
[root@scspa2385980001 harvest]#

[root@scspa2385980001 harvest]# ps aux | grep poller
root        28  0.0  0.0      0     0 ?        I<   May20   0:00 [edac-poller]
harvest  27762  2.5  1.1 1241188 93524 ?       Sl   13:56   0:59 bin/poller --poller F2240-127-26 --loglevel 2 --promPort 12991 --config /opt/harvest/harvest.yml --daemon
root     28165  0.0  0.0 221860  1100 pts/0    S+   14:36   0:00 grep --color=auto poller
[root@scspa2385980001 harvest]# 



[root@scspa2385980001 harvest]# bin/harvest start
Datacenter            Poller                PID        PromPort        Status              
+++++++++++++++++++++ +++++++++++++++++++++ ++++++++++ +++++++++++++++ ++++++++++++++++++++
umeng_aff300a         umeng_aff300a         28188      12990           running             
F2240-127-26          F2240-127-26          27762      12991           running             
+++++++++++++++++++++ +++++++++++++++++++++ ++++++++++ +++++++++++++++ ++++++++++++++++++++
[root@scspa2385980001 harvest]# 
[root@scspa2385980001 harvest]# 
[root@scspa2385980001 harvest]# ps aux | grep poller
root        28  0.0  0.0      0     0 ?        I<   May20   0:00 [edac-poller]
harvest  27762  2.5  1.1 1241188 91732 ?       Sl   13:56   1:01 bin/poller --poller F2240-127-26 --loglevel 2 --promPort 12991 --config /opt/harvest/harvest.yml --daemon
harvest  28188 14.6  1.0 1166880 85896 ?       Sl   14:37   0:00 bin/poller --poller umeng_aff300a --loglevel 2 --promPort 12990 --config ./harvest.yml --daemon
root     28197  0.0  0.0 221860  1096 pts/0    R+   14:37   0:00 grep --color=auto poller
[root@scspa2385980001 harvest]#
